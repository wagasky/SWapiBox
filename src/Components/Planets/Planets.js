import React from 'react';
import Card from '../Card/Card';
import { array, func } from 'prop-types';

const Planets = ({ data, handleFavorite }) => {
  
  const renderedCards = data.map( (planet, index) => {

    return (
      <Card 
        key={index}
        data={planet}
        handleFavorite={handleFavorite}
      />
    );
  });

  return (
    <div className="Planets">
      { renderedCards }
    </div>
  );
};

Planets.propTypes = {
  data: array,
  handleFavorite: func,
};

export default Planets;