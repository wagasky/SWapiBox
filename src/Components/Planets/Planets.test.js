/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import Planets from './Planets';
import { shallow } from 'enzyme';

describe('Planets', () => {

  let renderedComponent;
  let mockHandleFavorite;

  const mockData = [
      {"name":"Alderaan","terrain":"grasslands, mountains","population":"2000000000","climate":"temperate","residents":"Leia Organa\nBail Prestor Organa\nRaymus Antilles"},
      {"name":"Yavin IV","terrain":"jungle, rainforests","population":"1000","climate":"temperate, tropical","residents":"none"},
      {"name":"Hoth","terrain":"tundra, ice caves, mountain ranges","population":"unknown","climate":"frozen","residents":"none"},
    ]

  it('should match snapshot', () => {
    renderedComponent = shallow(<Planets data={mockData} handleFavorite={mockHandleFavorite} />);

    expect(renderedComponent).toMatchSnapshot();
  });

});