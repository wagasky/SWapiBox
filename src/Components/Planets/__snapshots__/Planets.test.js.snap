// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should match snapshot 1`] = `
<div
  className="Planets"
>
  <Card
    data={
      Object {
        "climate": "temperate",
        "name": "Alderaan",
        "population": "2000000000",
        "residents": "Leia Organa
      Bail Prestor Organa
      Raymus Antilles",
        "terrain": "grasslands, mountains",
      }
    }
    key="0"
  />
  <Card
    data={
      Object {
        "climate": "temperate, tropical",
        "name": "Yavin IV",
        "population": "1000",
        "residents": "none",
        "terrain": "jungle, rainforests",
      }
    }
    key="1"
  />
  <Card
    data={
      Object {
        "climate": "frozen",
        "name": "Hoth",
        "population": "unknown",
        "residents": "none",
        "terrain": "tundra, ice caves, mountain ranges",
      }
    }
    key="2"
  />
</div>
`;
